src/backend/file_chooser.rs

```diff
@@ -0,0 +1,293 @@
+use async_trait::async_trait;
+use futures_channel::{
+    mpsc::{Receiver, Sender},
+    oneshot,
+};
+use futures_util::{FutureExt, StreamExt};
+use std::cell::RefCell;
+use zbus::{dbus_interface, names::WellKnownName};
+
+use crate::{
+    backend::IMPL_PATH,
+    desktop::{
+        file_chooser::{Choice, FileFilter},
+        Response,
+    },
+    zvariant::{DeserializeDict, OwnedObjectPath, SerializeDict, Type},
+    WindowIdentifierType,
+};
+
+// Does not coincide with the one in desktop/file_chooser.rs
+#[derive(DeserializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+pub struct OpenFileOptions {
+    pub accept_label: Option<String>,
+    pub modal: Option<bool>,
+    pub multiple: Option<bool>,
+    pub directory: Option<bool>,
+    pub filters: Option<Vec<FileFilter>>,
+    pub current_filter: Option<FileFilter>,
+    pub choices: Option<Vec<Choice>>,
+}
+
+// Does not coincide with the one in desktop/file_chooser.rs
+#[derive(DeserializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+pub struct SaveFileOptions {
+    pub accept_label: Option<String>,
+    pub modal: Option<bool>,
+    pub multiple: Option<bool>,
+    pub filters: Option<Vec<FileFilter>>,
+    pub current_filter: Option<FileFilter>,
+    pub choices: Option<Vec<Choice>>,
+    pub current_name: Option<String>,
+    pub current_folder: Option<Vec<u8>>,
+    pub current_file: Option<Vec<u8>>,
+}
+
+#[derive(DeserializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+pub struct SaveFilesOptions {
+    // TODO Its in the xdp docs, but is it correct? See
+    // https://github.com/flatpak/xdg-desktop-portal/issues/938
```

Comment:
This is probably a docs issue, the handle_token doesn't make much sense here and should be dropped.

Reply:
Ill leave the comment and comment 
```rust
    // pub handle_token: Option<String>,
```